src/tokens/split_spaces.c

```diff
@@ -0,0 +1,78 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   split_spaces.c                                     :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: mcourtoi <mcourtoi@student.42.fr>          +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2023/03/27 20:50:26 by mcourtoi          #+#    #+#             */
+/*   Updated: 2023/03/27 21:06:38 by mcourtoi         ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "minishell.h"
+
+int	split_is_null(t_token_lst *const token_lst,
+		t_token **const token, t_str const *const tmp)
+{
+	if (ft_strlen((*token)->str))
+	{
+		if (tmp->next)
+		{
+			if (token_lst_add_after(token_lst, (*token), T_ARGUMENT, ""))
+				return (EXIT_FAILURE);
+			(*token) = (*token)->next;
+		}
+	}			
+	return (EXIT_SUCCESS);
+}
+
+int	add_spaces(t_token_lst *const token_lst, t_token **token,
+		t_str *tmp, char **split)
+{
+	if (ft_strlen((*token)->str))
```

Comment:
Same reflection, if you only intend to check whether `(*token)->str` is an empty string, only checking `*(*token)->str` would suffice.